client/src/AppBundle/Entity/Report/Document.php

```diff
@@ -343,6 +353,26 @@ public function isAdminDocument()
         return $this->isReportPdf() || $this->isTransactionDocument();
     }
 
+    public function supportingDocumentCanBeSynced()
+    {
+        return !$this->isReportPdf() && $this->getReport()->reportPdfHasBeenSynced();
+    }
+
+    public function getReportPdfSubmission()
+    {
+        if ($this->isReportPdf()) {
+            return $this->getReportSubmission();
+        }
+
+        foreach ($this->getReport()->getReportSubmissions() as $submission) {
+            if ($submission->hasReportPdf()) {
+                return $submission;
+            }
+        }
```

Comment:
I'm not sure this will handle resubmissions (after OPG has returned the report to the deputy) properly. It looks like it will always return the _first_ submission with a PDF when we probably want it to return the _previous_.

So we might need to do something more like:
```suggestion
        $previousSubmissions = array_filter($this->getReport()->getReportSubmissions(), function($submission) { $submission->createdOn <= $this->getReportSubmission()->createdOn; });
        foreach (array_reverse($previousSubmissions) as $submission) {
            if ($submission->hasReportPdf()) {
                return $submission;
            }
        }
```

Reply:
Ah I hadn't considered that - I've updated to use a filtered list now.